packages/types/src/envelope.ts

```diff
@@ -0,0 +1,69 @@
+import { SentryRequestType } from './request';
+import { SdkInfo } from './sdkinfo';
+import { Session, SessionAggregates } from './session';
+import { Outcome } from './transport';
+import { User } from './user';
+
+// Based on: https://develop.sentry.dev/sdk/envelopes/
+
+type CommonEnvelopeHeaders = {
+  dsn?: string;
+  sdk?: SdkInfo;
+};
+
+type CommonEnvelopeItemHeaders = {
+  length?: number;
+};
+
+/**
+ * 1st Item: Item headers
+ * 2nd Item: Item payload
+ */
+type BaseEnvelopeItem<ItemHeader extends { type: string }, Payload = unknown> = [
+  CommonEnvelopeItemHeaders & ItemHeader,
+  Payload,
+];
+
+type UnknownEnvelopeItem = BaseEnvelopeItem<{ type: '__unknown__' }>;
+
+type BaseEnvelope<
+  EnvelopeHeaders extends Record<string, unknown>,
+  EnvelopeItem extends BaseEnvelopeItem<{ type: string }>,
+> = {
+  headers: CommonEnvelopeHeaders & EnvelopeHeaders;
+  items: Array<EnvelopeItem | UnknownEnvelopeItem>;
+};
+
+export type EventEnvelopeItem = BaseEnvelopeItem<{ type: 'event' | 'transaction' }, Event>;
```

Comment:
This line cause a type error on node version 14.

Reply:
Huh interesting. Could you open a GitHub issue please with your tsconfig, package.json and some details about your setup?

Reply:
I will try to do that tomorrow, we now locked the `@sentry/types` to `6.17.7` in our package.json and that resolves the issue.

```
...
"@sentry/types": "6.17.7"
```

Reply:
Ah I think it's filed with https://github.com/getsentry/sentry-javascript/issues/4583. Woooops, thats on me. Will fix with https://github.com/getsentry/sentry-javascript/pull/4584